clients/src/main/java/org/apache/kafka/common/network/NetworkReceive.java

```diff
@@ -143,11 +146,14 @@ public long readFromReadableChannel(ReadableByteChannel channel) throws IOExcept
         }
         if (buffer != null) {
             int bytesRead = channel.read(buffer);
+            log.trace("readFromReadableChannel bytes read into buffer buffer {}", bytesRead);
             if (bytesRead < 0)
                 throw new EOFException();
             read += bytesRead;
         }
 
+        log.trace("readFromReadableChannel read {} bytes of data, is read complete {} with size.hasRemaining {}, "
+            + " buffer empty {}, buffer hasRemaining {}", read, complete(), size.hasRemaining(), buffer != null, buffer.hasRemaining());
```

Comment:
when buffer is null, the last argument runs into NPE.
So maybe we should have a separate variable to track the buffer.remaining() and default to 0 when the buffer is null.

Reply:
thanks for catching this, fixed